proton-j/src/main/java/org/apache/qpid/proton/engine/impl/EventImpl.java

```diff
@@ -217,6 +101,19 @@ public void dispatch(Handler handler)
             dispatch(children.next());
         }
     }
+    
+    @Override
+    public void redispatch(EventType as_type, Handler handler) throws HandlerException 
```

Comment:
A little unclear why this is needed, but ignoring that...it seems like using it inside a handler (as in the test) would lead to 'redispatch' to the current and child handlers before the original event is then subsequently given to any child handlers. Expected? Worth documenting?